modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -332,6 +332,49 @@ CV_EXPORTS_W bool haveImageReader( const String& filename );
  */
 CV_EXPORTS_W bool haveImageWriter( const String& filename );
 
+/** @brief To read Multi Page images on demand
+
+The ImageCollection class provides iterator API to read multi page images on demand. Create iterator
+to the collection of the images and iterate over the collection. Decode the necessary page with operator*.
+
+The performance of page decoding is O(1) if collection is increment sequentially. If the user wants to access random page,
+then the time complexity is O(n) because the collection has to be reinitialized every time in order to go correct page.
```

Comment:
O(n) is not fully true. For TIFF `TIFFReadDirectory() and readHeader()` are called, but not full decoding pipeline.

Reply:
how about "Complexity is O(n) because the collection has to be reinitialized every time in order to go to the correct page. However, the intermediate pages are not decoded during the process, so typically it's quite fast"?